knife/lib/chef/utils/licensing_handler.rb

```diff
@@ -30,9 +30,16 @@ def install_sh_url
 
       class << self
         def validate!
-          license_keys = ChefLicensing::LicenseKeyFetcher.fetch
-
-          return new(nil, nil) if license_keys.blank?
+          license_keys = begin
+                           ChefLicensing::LicenseKeyFetcher.fetch
+                         # If the env is airgapped or the local licensing service is unreachable,
+                         # the licensing gem will raise ChefLicensing::RestfulClientConnectionError.
+                         # In such cases, we are assuming the license is not available.
+                         rescue ChefLicensing::RestfulClientConnectionError
+                           []
+                         end
+
+          return new(nil, nil) if license_keys&.empty?
```

Comment:
@ashiqueps I'm assuming `license_keys` would have only been `nil` if an exception occurred. Otherwise, you need `|| license_keys.nil?` if `nil` is still possible.